impeller/compiler/shader_lib/impeller/texture.glsl

```diff
@@ -143,6 +143,15 @@ vec4 IPSampleDecal(sampler2D texture_sampler, vec2 coords) {
   return texture(texture_sampler, coords);
 }
 
+/// Sample a texture with decal tile mode.
+f16vec4 IPHalfSampleDecal(f16sampler2D texture_sampler, f16vec2 coords) {
+  if (any(lessThan(coords, f16vec2(0.0hf))) ||
+      any(greaterThanEqual(coords, f16vec2(1.0hf)))) {
+    return f16vec4(0.0hf);
+  }
```

Comment:
Not clamp?

Why are we truncating to zero always here?

Reply:
this is how decal works...

Reply:
Ahh I missed that